test/core/event_engine/test_suite/tests/dns_test.cc

```diff
@@ -107,144 +96,67 @@ MATCHER(StatusCodeEq, "") {
   return std::get<0>(arg).code() == std::get<1>(arg);
 }
 
-// Copied from tcp_socket_utils_test.cc
-// TODO(yijiem): maybe move those into common test util
-EventEngine::ResolvedAddress MakeAddr4(const uint8_t* data, size_t data_len,
-                                       int port) {
-  EventEngine::ResolvedAddress resolved_addr4;
-  sockaddr_in* addr4 = reinterpret_cast<sockaddr_in*>(
-      const_cast<sockaddr*>(resolved_addr4.address()));
-  memset(&resolved_addr4, 0, sizeof(resolved_addr4));
-  addr4->sin_family = AF_INET;
-  GPR_ASSERT(data_len == sizeof(addr4->sin_addr.s_addr));
-  memcpy(&addr4->sin_addr.s_addr, data, data_len);
-  addr4->sin_port = htons(port);
-  return EventEngine::ResolvedAddress(
-      reinterpret_cast<sockaddr*>(addr4),
-      static_cast<socklen_t>(sizeof(sockaddr_in)));
-}
-
-EventEngine::ResolvedAddress MakeAddr6(const uint8_t* data, size_t data_len,
-                                       int port) {
-  EventEngine::ResolvedAddress resolved_addr6;
-  sockaddr_in6* addr6 = reinterpret_cast<sockaddr_in6*>(
-      const_cast<sockaddr*>(resolved_addr6.address()));
-  memset(&resolved_addr6, 0, sizeof(resolved_addr6));
-  addr6->sin6_family = AF_INET6;
-  GPR_ASSERT(data_len == sizeof(addr6->sin6_addr.s6_addr));
-  memcpy(&addr6->sin6_addr.s6_addr, data, data_len);
-  addr6->sin6_port = htons(port);
-  return EventEngine::ResolvedAddress(
-      reinterpret_cast<sockaddr*>(addr6),
-      static_cast<socklen_t>(sizeof(sockaddr_in6)));
-}
-
-#define EXPECT_UNKNOWN_ERROR(result) \
-  EXPECT_EQ((result).status().code(), absl::StatusCode::kUnknown)
+#define EXPECT_STATUS(result, status_code) \
```

Comment:
I think that instead of having this macro, we should just use something like `EXPECT_EQ(status, absl::UnavailableError("..."))`.  That will both make the code clearer and will check the error message as well as the status code.

Reply:
Done.